src/scribe_data/wiktionary/wikitionary_utils.py

```diff
@@ -0,0 +1,177 @@
+        try_old_dump (function): A function to validate if the dump file exists.
+
+    Returns:
+        str: The closest available dump file date (as a string).
+        None: If no suitable dump is found.
+    """
+    available_dates = []
+    target_date = parse_date(target_entity)
+    closest_date = None
+    closest_diff = None
+
+    if target_date:
+        for i in other_old_dumps:
+            if i == "..":
+                continue
+            try:
+                if try_old_dump(i):
+                    available_dates.append(i)
+                    current_date = parse_date(i)
+                    diff = abs((current_date - target_date).days)
+
+                    if closest_diff is None or diff < closest_diff:
+                        closest_date = i
+                        closest_diff = diff
+
+                    if current_date >= target_date:
+                        break
+            except requests.exceptions.HTTPError:
+                pass
+        return closest_date
+
+
+def download_wiki_lexeme_dump(target_entity="latest-lexemes"):
+    """
+    Downloads a Wikimedia lexeme dump based on the specified target entity or date.
+
+    Args:
+        target_entity (str, optional): The target dump to download. Defaults to "latest-lexemes".
+            - If "latest-lexemes", downloads the latest dump.
+            - If a valid date (e.g., YYYY/MM/DD), attempts to download the dump for that date.
+
+    Returns:
+        str: The URL of the requested or closest available dump.
+        None: If no suitable dump is found or the request fails.
+    """
+    base_url = "https://dumps.wikimedia.org/wikidatawiki/entities"
+
+    def try_old_dump(target_entity):
+        """
+        Checks if the specified dump file exists for a target entity.
+
+        Args:
+            target_entity (str): The target entity or date folder to check.
+
+        Returns:
+            str: The URL of the dump file if it exists.
+            None: If the dump file does not exist.
+        """
+        entity_url = f"{base_url}/{target_entity}/"
+        entity_response = requests.get(entity_url)
+        entity_response.raise_for_status()
+        dump_filenames = re.findall(r'href="([^"]+)"', entity_response.text)
+
+        fileurl = f"wikidata-{target_entity}-lexemes.json.bz2"
+        if fileurl in dump_filenames:
+            return f"{base_url}/{target_entity}/{fileurl}"
+
+    if target_entity != "latest-lexemes":
+        try:
+            if parse_date(target_entity):
+                target_entity = target_entity.replace("/", "").replace("-", "")
+                return try_old_dump(target_entity)
+
+        except requests.exceptions.HTTPError as http_err:
+            print(
+                f"HTTP error occurred: {http_err} Status code: {http_err.response.status_code}"
+            )
+            print("We could not found for your requested lexeme dump wikitionary.")
```

Comment:
```suggestion
            print("We could not find your requested wikidata lexeme dump.")
```
nit - wording suggestion